runners/flink/src/main/java/org/apache/beam/runners/flink/translation/types/CoderTypeSerializer.java

```diff
@@ -139,24 +140,28 @@ public TypeSerializerConfigSnapshot snapshotConfiguration() {
 
   @Override
   public CompatibilityResult<T> ensureCompatibility(TypeSerializerConfigSnapshot configSnapshot) {
-    if (configSnapshot instanceof CoderTypeSerializerConfigSnapshot) {
-      if (coder.equals(((CoderTypeSerializerConfigSnapshot<?>) configSnapshot).coder)) {
-        return CompatibilityResult.compatible();
-      }
+    if (snapshotConfiguration().equals(configSnapshot)) {
+      return CompatibilityResult.compatible();
     }
     return CompatibilityResult.requiresMigration();
   }
 
   /**
-   *  TypeSerializerConfigSnapshot of CoderTypeSerializer.
+   *  TypeSerializerConfigSnapshot of CoderTypeSerializer. This uses the class name of the
+   *  {@link Coder} to determine compatibility. This is a bit crude but better than using
+   *  Java Serialization to (de)serialize the {@link Coder}.
```

Comment:
In Flink, there has been a trend to move as far away from Java Serialization as possible because it can lead to problems in deserialisation when the code is changed.

Simply relying on the class name is also not the best solution, I admit. At some point there might be a Coder versioning API in Beam. @kennknowles Do you have any comments? I think you once mentioned that you had some work on coder versioning.